### PR TITLE
Simplify the gradient clipping code

### PR DESCRIPTION
**Pull request by datumbox**
As discussed at https://github.com/pytorch/vision/pull/4874#discussion_r744539820

### CODE REVIEW

**Review by sallysyw — APPROVED** [review]
LGTM! Thanks!
